packages/graphql/tests/performance/translation/translation-performance.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { gql } from "apollo-server";
+import path from "path";
+import { Neo4jGraphQL } from "../../../src";
+import { collectTests } from "../utils/collect-test-files";
+import { translateQuery } from "./utils/translate-query";
+import type * as Performance from "../types";
+import { colorText, TTYColors } from "../utils/formatters/color-tty-text";
+import { typeDefs } from "../typedefs";
+type TranslateTestConfig = {
+    runs: number;
+    neoSchema: Neo4jGraphQL;
+    sync: boolean;
+};
+
+async function runTest(test: Performance.TestInfo, config: TranslateTestConfig) {
+    const query = gql(test.query);
+    const schema = config.neoSchema;
+    const THRESHOLD = 1; // Millisecond per translation
+    const totalThreshold = THRESHOLD * config.runs;
+
+    // Warmup
+    await translateQuery(schema, query);
```

Comment:
Just a thought. Perhaps a warmup isn't really necessary for this since it's just code. Alternatively, if we want to remove outliers then we should probably cull the quickest and slowest N results.

Reply:
For code is usually recommended a warmup as the JIT compilation and cpu shenanigans may skew the results and this is way simpler than tracking all of the results, which we may want to do if we find this is still too unrealiable

Reply:
Sounds good!